manifests/fcontext.pp

```diff
@@ -63,50 +63,42 @@
 # }
 #
 define selinux::fcontext (
-  $destination = '',
+  $pathname,
+  $destination = undef,
   $context     = '',
-  $pathname    = '',
   $filetype    = false,
-  $filemode    = '',
+  $filemode    = undef,
   $equals      = false,
-  $policy      = 'targeted',
 ) {
-  Exec {
-    path => '/bin:/sbin:/usr/bin:/usr/sbin',
-  }
-  if $pathname == '' {
-    fail('pathname must not be empty')
-  }
 
-  if ( $equals == false ) and ( $pathname == '' ) {
-    fail('pathname must not be empty')
+  if !defined(Class['selinux']) {
+    fail('You must include the selinux base class before using any selinux defined resources')
```

Comment:
If the base class was heavier, I'd agree. As it stands, the user really only is specifying the mode `selinux` is going to be in, so it's not too much of a stretch to provide a sane default and attempt to declare the class if it hasn't been before.